tests/steps/query_balance_test.go

```diff
@@ -0,0 +1,117 @@
+package steps
+
+import (
+	"context"
+	"errors"
+	"math/big"
+	"testing"
+
+	"github.com/cucumber/godog"
+	"github.com/make-software/casper-go-sdk/casper"
+	"github.com/make-software/casper-go-sdk/rpc"
+
+	"github.com/stormeye2000/cspr-sdk-standard-tests-go/tests/utils"
+)
+
+// Step Definitions for the query_balance.feature
+func TestFeaturesQueryGetBalance(t *testing.T) {
+	utils.TestFeatures(t, "query_balance.feature", InitializeQueryGetBalance)
+}
+
+func InitializeQueryGetBalance(ctx *godog.ScenarioContext) {
+	var sdk casper.RPCClient
+	var queryGetBalanceResult rpc.QueryBalanceResult
+	var queryGetBalanceJson string
+
+	ctx.Before(func(ctx context.Context, _ *godog.Scenario) (context.Context, error) {
+		utils.ReadConfig()
+		sdk = utils.GetSdk()
+		return ctx, nil
+	})
+
+	ctx.Step(`^that a query balance is obtained by main purse public key$`, func() error {
+		faucetKey, err := casper.NewED25519PrivateKeyFromPEMFile("../../assets/net-1/faucet/secret_key.pem")
+
+		if err == nil {
```

Comment:
I'd get annoyed with all the code in an if block
I'd do:
`if err !=nil {
    return err
}
....
`